api/src/libs/migrateBusinessesFromUsers.ts

```diff
@@ -0,0 +1,45 @@
+import { BusinessesDataClient, UserDataClient } from "@domain/types";
+import { LogWriterType } from "@libs/logWriter";
+import { processBusinessesMigration } from "@libs/migrationUtils";
+import { UserData } from "@shared/userData";
+
+export const migrateBusinessesFromUsers = async (
+  userDataClient: UserDataClient,
+  businessesDataClient: BusinessesDataClient,
+  logger: LogWriterType
+): Promise<void> => {
+  const processBatch = async (users: UserData[]): Promise<void> => {
+    for (const userData of users) {
+      logger.LogInfo(`Starting migration for userId: ${userData.user.id}`);
+      const businessesData = userData.businesses;
+      if (businessesData && Object.keys(businessesData).length > 0) {
+        await processBusinessesMigration(
+          userData,
+          businessesData,
+          userDataClient,
+          businessesDataClient,
+          logger
+        );
+      } else {
+        logger.LogError(`Businesses not found for user ID: ${userData.user.id}`);
+      }
+    }
+  };
+
+  const migrateUsersInPages = async (lastEvaluatedKey?: string): Promise<void> => {
+    const usersWithBusinesses = await userDataClient.getUsersWithBusinesses(lastEvaluatedKey);
+    await processBatch(usersWithBusinesses.users);
+
+    if (usersWithBusinesses.lastEvaluatedKey) {
+      await migrateUsersInPages(usersWithBusinesses.lastEvaluatedKey);
+    }
+  };
+
+  try {
+    await migrateUsersInPages();
```

Comment:
to confirm, when `lastEvaluatedKey` is initially not provided, the search, getUsersWithBusinesses will return a value?

Reply:
The function will return a value even if lastEvaluatedKey is not provided initially. When not defined, the query starts from the beginning of the dataset. If additional results are available for pagination, the lastEvaluatedKey from the query response will be returned. This ensures that the function handles both the initial case (no key required) and subsequent calls when pagination is needed.